install.sh

```diff
@@ -30,16 +30,61 @@ RED='\e[1;91m';    # Use Light Red for errors.
 YELLOW='\e[1;33m'; # Use Yellow for informational messages.
 GREEN='\e[1;32m';  # Use Green for success messages.
 BLUE='\e[1;34m';   # Use Blue for intrafunction messages.
-GRAY='\e[1;37m';   # Use Gray for program output.
+GRAY='\e[0;37m';   # Use Gray for program output.
 MAGENTA='\e[1;35m';
 RESET='\e[0m'
 
-echo -e "${GREEN}Welcome to Chromebrew!${RESET}\n"
+# skip all checks if running on a docker container
+grep ':/docker' /proc/self/cgroup &> /dev/null && CREW_FORCE_INSTALL=1
+
+# reject crostini
+if [[ -d /opt/google/cros-containers && "${CREW_FORCE_INSTALL}" != '1' ]]; then
+  echo -e "${RED}Crostini containers are not supported by Chromebrew :/${RESET}"
+  echo -e "${YELLOW}Run 'curl -Ls git.io/vddgY | CREW_FORCE_INSTALL=1 bash' to perform install anyway${RESET}"
+  exit 1
+fi
+
+# disallow non-stable channels Chrome OS
+if [ -f /etc/lsb-release ]; then
+  if [[ ! "$(< /etc/lsb-release)" =~ CHROMEOS_RELEASE_TRACK=stable-channel$'\n' && "${CREW_FORCE_INSTALL}" != '1' ]]; then
+    echo -e "${YELLOW}The beta, dev, and canary channel are unsupported by Chromebrew${RESET}"
+    echo -e "${YELLOW}Run 'curl -Ls git.io/vddgY | CREW_FORCE_INSTALL=1 bash' to perform install anyway${RESET}"
+    exit 1
+  fi
+else
+  echo -e "${YELLOW}Unable to detect system information, installation will continue.${RESET}"
+fi
+
+echo -e "\n${GREEN}Welcome to Chromebrew!${RESET}\n"
 if [ "${EUID}" == "0" ]; then
   echo -e "${RED}Chromebrew should not be installed or run as root.${RESET}"
   exit 1;
 fi
 
+# prompt user to enter the sudo password if it set
+# if the PASSWD_FILE specified by chromeos-setdevpasswd exist, that means a sudo password is set
+if [[ "$(< /usr/sbin/chromeos-setdevpasswd)" =~ PASSWD_FILE=\'([^\']+) ]] && [ -f "${BASH_REMATCH[1]}" ]; then
+  echo -e "${BLUE}Please enter the developer mode password${RESET}"
+  # reset sudo timeout
+  sudo -k
+  sudo /bin/true
```

Comment:
Are 2 sudo lines necessary?

Reply:
The first `sudo` is used to reset the password timeout, so we can confirm that a password prompt will be shown in next sudo command

A password prompt will show in the second `sudo`, then the following `sudo` will not ask for the password (unless the `sudo` password timestamp expired)